core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableSort.java

```diff
@@ -50,7 +50,8 @@ public static EnumerableSort create(RelNode child, RelCollation collation,
     final RelTraitSet traitSet =
         cluster.traitSetOf(EnumerableConvention.INSTANCE)
             .replace(collation);
-    return new EnumerableSort(cluster, traitSet, child, collation, offset,
+    return new EnumerableSort(cluster, traitSet, child,
+        (RelCollation) traitSet.getTrait(collation.getTraitDef()), offset,
```

Comment:
This is not needed.

Reply:
It's needed. There's an assert in Sort.java line 94, which checks the collation is indeed the same trait object within RelTraitSet. I am not sure why we would need such check. For now I just pass in the collation trait object within traitSet to make it happy.

Reply:
We still don't need it. Because there is another bug causing it:
https://issues.apache.org/jira/browse/CALCITE-3949

Reply:
Sure, I can fix this after 3949 is merged.

Reply:
fixed